src/Dialect/ONNX/DialectBuilder.cpp

```diff
@@ -134,6 +134,19 @@ Value OnnxBuilder::matmul(Type Y, Value A, Value B, bool useGemm) const {
   return createOpAndInferShapes<ONNXMatMulOp>(toTensor(Y), aValue, bValue);
 }
 
+Value OnnxBuilder::max(ValueRange inputs) const {
+  assert(inputs.size() >= 2 && "Expect at least two inputs");
+  Type elementType = getElementType(inputs[0].getType());
+  assert(llvm::all_of(inputs, [elementType](Value v) {
+    return (v.getType().cast<ShapedType>().getElementType() == elementType);
+  }) && "All inputs must have the same element type");
+  Type outputType = inputs[0].getType();
+  for (uint64_t i = 1; i < inputs.size(); ++i)
+    outputType = mlir::OpTrait::util::getBroadcastedType(
+        toTensor(outputType), inputs[i].getType());
+  return createTypedOpAndInferShapes<ONNXMaxOp>(toTensor(outputType), inputs);
```

Comment:
given that createTypedOpAndInferShapes calls inferShapes, will it work to just pass an unranked tensor output type and skip the calls to getBroadcastedType?

Reply:
Good catch. Will check this.